### ISSUE TITLE
`_requestAllPages` pagination is broken

### ISSUE THREAD

**Issue by connorjclark** [github-tools/github#569]
https://github.com/github-tools/github/blob/943ac8a7d36dbac6c0848bee9a5ca64dc229f807/lib/Requestable.js#L259

This bit: 
```
typeof options.page !== 'number'
```

can never be true, as the api functions (well, at least the one I need, `listStarredRepos`) don't allow for passing in options. Besides, it doesn't make much sense, I think it should be negated?

I have this line in my build script to bypass this issue:
```
sed -i '' "s/typeof options.page !== 'number'/false/g" dist/scripts/GitHub.bundle.js
```

**Comment by tamer1an** [comment]
Yes it seems i can only get 100 starred repos

**Comment by tamer1an** [comment]
Here is the fix: https://github.com/github-tools/github/pull/575

**Comment by j-rewerts** [comment]
I'll take look at this soon. thanks!

**Comment by j-rewerts** [comment]
This should be resolved in [3.2.3](https://www.npmjs.com/package/github-api). Please try and confirm.
